### DISCUSSION TITLE
How can I change mutagen settings in predefined warden enviroment?

### DISCUSSION THREAD

**Discussion by mykola-yatsenko** [wardenenv/warden#782] [Help & Support]
Hi there!

I am a fresher in using Warden.

I found it like an awesome develop tool! 

**Steps to reproduce:**

1. I had initiated warden project with predefined enviroment [Magento 2]
2. I had installed Opencart 2 

When I made some code changes in a file:

`pub/catalog/controller/product/product.php`
Then I reload web page. (I had checked throgh warden shell as well)  I see no difference. 

-------

When I made some code changes  in a file 
`pub/system/engine/model.php`
Then I reload web page (I had checked throgh warden shell as well) . I see the difference.

Probably,  Warden Magento 2 Enviroment Mutagen   is ignoring files in a **catalog/...** folder
I had tried to change several other files in **catalog** folder, and observed no difference in the browser.

On other hand, I found nothing about **catalog** folder in [mutagen config file here](https://github.com/wardenenv/warden/blob/main/environments/magento2/magento2.mutagen.yml) 

1. How can I customize mutagen configuration?
2. Does anyone has ideas how to figure it out?
--------

I have .warden/warden-env.yml 
`version: "3.5"
services:
  nginx:
    - ./.warden/nginx/custom.conf:/etc/nginx/default.d/custom.conf
`

I had tried to create file as well
.warden/warden-env.mutagen.yml
`---
sync:
  defaults:
    mode: two-way-resolved
    watch:
      pollingInterval: 10
    ignore:
      vcs: false
      paths:
        # Root .git folder
        - "/.git/"

        # System files
        - ".DS_Store"
        - "._*"

    permissions:
      defaultFileMode: "0644"
      defaultDirectoryMode: "0755"
`

I am not sure that it works.

**Comment by bap14** [comment]
@mykola-yatsenko It doesn't look like it's documented, but you can create a `.warden/mutagen.yml` file in your project and it will be used instead of the default Magento environment mutagen configuration file. You can start by copying the [mutagen file from the Magento 2 template](https://github.com/wardenenv/warden/blob/main/environments/magento2/magento2.mutagen.yml) or just start fresh. Then just restart your environment and it should pick up your new configuration.
